src/Tribe/Attendee_Registration/Template.php

```diff
@@ -102,6 +98,41 @@ public function setup_context( $posts ) {
 
 	}
 
+	/**
+	 * Returns whether or not the user is on a custom attendee registration page.
+	 *
+	 * @since TBD
+	 *
+	 * @return bool
+	 */
+	public function is_on_custom_ar_page() {
+		global $wp_query, $post;
+
+		$ar_page_slug = tribe( 'tickets.attendee_registration' )->get_slug();
+
+		// Check for custom AR page by page slug.
+		$on_custom_page = ! empty( $wp_query->query_vars['pagename'] )
+			&& $ar_page_slug === $wp_query->query_vars['pagename'];
+
+		if ( ! $on_custom_page ) {
+			return false;
+		}
+
+		$uses_shortcode = ! empty( $post->post_content )
+			&& has_shortcode( $post->post_content, 'tribe_attendee_registration' );
+
+		if ( $on_custom_page && $uses_shortcode ) {
```

Comment:
checking `on_custom_page` here is redundant.
```suggestion
		if ( $uses_shortcode ) {
```